### ISSUE TITLE
NTK-based exllama support potential

### ISSUE THREAD

**Issue by Midaychi** [oobabooga/text-generation-webui#2948]
https://github.com/turboderp/exllama/pull/118

This hasn't been vetted/merged yet but in practice, it seems to unlock the context of un-finetuned models based on the scaling alpha value and does it with minimal perplexity loss.

Ironically superhot-tuned models fair worse with it.

Might be worth checking out, or at least keeping an eye on?

**Comment by Panchovix** [comment]
I have a PR already done to implement NTK RoPE Extension, if turbo approved the PR.

![image](https://github.com/oobabooga/text-generation-webui/assets/19153797/6a6b8956-46ca-4d01-aedc-c3b0f3f69748)

**Comment by Midaychi** [comment]
a

**Comment by Panchovix** [comment]
Here's my fork if you want to try in the meanwhile, you have to apply the exllama PR as well.

https://github.com/Panchovix/text-generation-webui

**Comment by Midaychi** [comment]
I manually tried it and it seems to work really well
 Even on an un fine-tuned 7b model.
I don't know if your or is gonna get merged though; turboderp's really actively venting and sus about it.

**Comment by Ph0rk0z** [comment]
It certainly doesn't break anything. I don't have superHOT for 65b and supposedly perplexity is bad on it before you reach 2048. For my use case (chat) this is better. The other alternative is smaller models.

I think everything is all merged now.

**Comment by Midaychi** [comment]
Yeah get ready for people to get confused at what the extra dial is or why their superhot models is performing worse with it.

Not blaming people but stuff is moving pretty fast. People are probably still just now finding out about rope stuff from YouTubers and coming in going qtf

**Comment by Ph0rk0z** [comment]
I am barely trying this stuff today as I've been training audio models. Added an extra dial to the UI for myself too :P

**Comment by Panchovix** [comment]
@Midaychi turbo merged, so I will do the PR now for ooba.

**Comment by Panchovix** [comment]
PR is here: https://github.com/oobabooga/text-generation-webui/pull/2955
